src/main/php/rdbms/Peer.class.php

```diff
@@ -257,7 +257,7 @@ public function iteratorFor(SQLExpression $criteria, $buffered= true) {
     $rs= $criteria->executeSelect($this->getConnection(), $this, $jp, $buffered);
 
     // if this is a projection, it does no matter if it's a join or not
-    if ($criteria->isProjection()) return new ResultIterator($rs, 'Record');
+    if ($criteria->isProjection()) return new ResultIterator($rs, '\rdbms\Record');
```

Comment:
> Maybe @thekid can answer this?

Sure 😄 

* * * 

The XP framework aims to support the PHP way of fully qualifying class names in any place the dotted names are supported. This is because PHP 5.5's *class* constants reads so nicely in these situations:

```php
$inject->bind('com.example.Test', new TestEverything());  // Dotted
$inject->bind(Test::class, new TestEverything());         // Class const w/ use statement
```

However, the preferred way for writing class literals inside strings is the dotted form. Using backslashes in strings can have unwanted side-effects, plus it's visually a problem compared to fully qualified class names: Inside strings, the leading backslash should not be used.